pkg/crc/preflight/preflight_checks_darwin.go

```diff
@@ -182,3 +185,27 @@ func addFileWritePermissionToUser(filename string) error {
 
 	return nil
 }
+
+func removeCRCHyperkitMachine() error {
+	err := os.RemoveAll(filepath.Join(constants.MachineInstanceDir, "crc"))
+	if err != nil {
+		logging.Debugf("Failed to remove hyperkit machine directory: %v", err)
+		return err
+	}
+	return nil
```

Comment:
windows/linux already have code doing that, have you considered moving this to `preflight_checks_common.go` since we want to do it for all platforms? I haven't looked if it's doable.

Reply:
Good idea, can create a `removeCRCMachineDir()` in `preflight_checks_common.go`